cmd/dp-deployer/main.go

```diff
@@ -117,7 +116,33 @@ func main() {
 	case <-ctx.Done():
 		log.Event(ctx, "context done", log.INFO)
 	}
-	wg.Wait()
+
+	log.Event(ctx, fmt.Sprintf("shutdown with timeout: %s", cfg.GracefulShutdownTimeout), log.INFO)
+	shutdownContext, cancel := context.WithTimeout(context.Background(), cfg.GracefulShutdownTimeout)
+
+	go func() {
+
+		// Shutdown HTTP server
+		log.Event(shutdownContext, "closing http server", log.INFO)
+		if err := httpServer.Shutdown(ctx); err != nil {
+			log.Event(shutdownContext, "failed to gracefully close http server", log.ERROR, log.Error(err))
+		}
+		log.Event(ctx, "http server gracefully closed ", log.INFO)
+
+		// Stop healthcheck
+		hc.Stop()
+
+		e.Close()
```

Comment:
does this return an error? If so probably best to check / log it out if its not nil. Otherwise this is fine.

Reply:
No `Close` does not return anything only logs